jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -160,11 +160,23 @@ public ContextHandler(Handler.Container parent, String contextPath)
             addAliasCheck(new SymlinkAllowedResourceAliasChecker(this));
     }
 
+    @Override
+    public void setServer(Server server)
+    {
+        super.setServer(server);
+        _mimeTypes.setWrapped(server.getMimeTypes());
```

Comment:
Does this mean `_mimeTypes` isn't available to modify until the server is started? (as the `setServer(Server)` call usually only happens in a doStart() call.

Reply:
No,
all modifications go into the context wrapper, before or after the wrapped types are set.

It just means that the defaults set in the server are not visible until the server is set.   
So context MimeTypes will initially look empty, but if you set some types, you will see exactly only those types... until the server is set, at which time you will also see all the server types... except the context types always override server types.